examples/async/index.js

```diff
@@ -1,15 +1,43 @@
 import 'babel-polyfill'
 import React from 'react'
-import { render } from 'react-dom'
+import ReactDOM from 'react-dom'
 import { Provider } from 'react-redux'
-import App from './containers/App'
 import configureStore from './store/configureStore'
 
 const store = configureStore()
+const rootEl = document.getElementById('root')
 
-render(
-  <Provider store={store}>
-    <App />
-  </Provider>,
-  document.getElementById('root')
-)
+let render = () => {
+  const App = require('./containers/App').default
```

Comment:
dynamic modules are not officially supported since the loader spec is still a work in progress.
webpack v2 supports the ES2015 draft proposal, ie:

```js
let render = System.import('./containers/App').then(App => {
  ReactDOM.render(
    <Provider store={store}>
      <App />
    </Provider>,
    rootEl
  )
})
```

i believe this is also supported in babel, using a polyfill and system module formatter